### PR TITLE
feat: shortcut to extended view

### PR DESCRIPTION
**Pull request by simonheys**
- Adds an 'Extended view' option to the Settings screen which opens the extension in a browser tab
- When in a tab, there are responsive horizontal margins (configurable, looks v weird without these)
- Handles the case where an external action is received to focus the extension if running in a tab

Screenshot 2022-06-13 at 16 44 55

Screenshot 2022-06-13 at 16 55 00

Screenshot 2022-06-13 at 16 55 08

### CODE REVIEW

File: packages/extension/src/ui/features/browser/tabs.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import browser from "webextension-polyfill"
+
+export const openExtensionInTab = async () => {
+  const url = browser.runtime.getURL("index.html")
+  const tab = await browser.tabs.create({ url })
+  return tab
+}
+
+export const extensionIsInTab = async () => {
+  return Boolean(await browser.tabs.getCurrent())
```
Comment (janek26) [inline packages/extension/src/ui/features/browser/tabs.ts:11]: sure getCurrent works as you expect? I would expect that it returns whatever tab is currently open by the window, eg google

File: packages/extension/src/ui/features/settings/SettingsScreen.tsx [hunk 2]
```diff
@@ -70,75 +74,91 @@ const Footer = styled.div`
...
+            <ArrowForwardIosIcon fontSize="inherit" />
+          </Title>
+        </SettingsItem>
+        <hr />
+        {!extensionIsInTab && (
+          <>
+            <SettingsItem to={routes.settings()} onClick={openExtensionInTab}>
+              <Title>
+                <span>Extended view</span>
+                <OpenInFullIcon fontSize="inherit" />
```
[the remark this answers is not in the document]
Comment (simonheys) [reply packages/extension/src/ui/features/settings/SettingsScreen.tsx]: Changed to same chevron icon as the others

File: packages/extension/src/ui/features/actions/ActionScreen.tsx [hunk 3]
```diff
@@ -1,12 +1,14 @@
-import { FC, useCallback } from "react"
+import { FC, useCallback, useEffect } from "react"
 import { useNavigate } from "react-router-dom"
+import browser from "webextension-polyfill"
```
[the remark this answers is not in the document]
Comment (simonheys) [reply packages/extension/src/ui/features/actions/ActionScreen.tsx]: Good catch - changed

File: packages/extension/src/ui/features/browser/tabs.ts (hunk 1 above)
Comment (simonheys) [reply packages/extension/src/ui/features/browser/tabs.ts:11]: https://developer.chrome.com/docs/extensions/reference/tabs/#method-getCurrent

> Gets the tab that this script call is being made from

It does seem to work as expected. I think ideally there would be some UI tests around this feature, not sure if that is possible with web extensions?

Comment (janek26) [reply packages/extension/src/ui/features/browser/tabs.ts:11]: ohh, cool to know!

e2e test are to be explored

**Review by janek26 — APPROVED** [review]

**Pull request merged**